pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -476,6 +476,13 @@ type CephObjectRealmList struct {
 
 // ObjectRealmSpec represent the spec of an ObjectRealm
 type ObjectRealmSpec struct {
+	Pull PullSpec `json:"pull,omitempty"`
+}
+
+type PullSpec struct {
+	Endpoint  string `json:"endpoint"`
+	AccessKey string `json:"accessKey,omitempty"`
```

Comment:
We can't add the access key and secret in plain text in the CR. There are a couple of approaches:
1. Have a CR setting, which is the name of the secret where the access key and secret key can be found in the cluster namespace. The admin would be required to create this k8s secret with that name before creating the realm CR.
1. Use a predefined secret name based on the name of the realm. For example, rook could expect a secret named `rook-ceph-object-realm-<name>-endpoint`, and have two keys: `accessKey` and `secretKey`.

I would lean towards the latter approach and just document the expected secret name.

Reply:
Approach number one seems more flexible as Rook will simply go and fetch whatever was requested by the admin.
If nothing is found, the controller will requeue until the Secret is found.